packages/nimble-components/src/select/index.ts

```diff
@@ -39,61 +88,801 @@ export class Select extends FoundationSelect implements ErrorPattern {
+            case keyEnd: {
+                e.preventDefault();
+                break;
+            }
+            case keyEnter: {
+                e.preventDefault();
+                if (
+                    this.filteredOptions.length === 0
+                    || this.filteredOptions.every(o => o.disabled)
+                ) {
+                    return false;
+                }
+                this.updateSelectedIndexFromFilteredSet();
+                this.open = !this.open;
+                if (!this.open) {
+                    this.focus();
+                }
+                break;
+            }
+            case keyEscape: {
+                // clear filter as update to "selectedIndex" will result in processing
+                // "options" and not "_options"
+                this.filter = '';
+                if (this.committedSelectedOption) {
+                    this.clearSelection();
+                    this.selectedIndex = this._options.indexOf(
+                        this.committedSelectedOption
+                    );
+                }
+                if (this.collapsible && this.open) {
+                    e.preventDefault();
+                    this.open = false;
+                }
+                // reset 'selected' state otherwise the selected state doesn't stick.
+                const selectedOption = this._options[this.selectedIndex];
+                if (selectedOption) {
+                    selectedOption.selected = true;
+                }
+                this.focus();
+                break;
+            }
+            case keyTab: {
+                if (this.collapsible && this.open) {
+                    e.preventDefault();
+                    this.open = false;
+                }
+
+                return true;
+            }
+
+            default: {
+                break;
+            }
+        }
+
+        if (!this.open && this.indexWhenOpened !== this.selectedIndex) {
+            this.updateValue(true);
+            this.indexWhenOpened = this.selectedIndex;
+        }
+
+        return !(key === keyArrowDown || key === keyArrowUp);
+    }
+
+    /**
+     * Updates the proxy value when the selected index changes.
+     *
+     * @param prev - the previous selected index
+     * @param next - the next selected index
+     *
+     * @internal
+     */
+    public override selectedIndexChanged(
+        prev: number | undefined,
+        next: number
     ): void {
-        if (this.region && this.control) {
-            this.region.anchorElement = this.control;
+        super.selectedIndexChanged(prev, next);
+        this.updateValue();
+    }
+
+    /**
+     * Synchronize the `aria-disabled` property when the `disabled` property changes.
+     *
+     * @param prev - The previous disabled value
+     * @param next - The next disabled value
+     *
+     * @internal
+     */
+    public override disabledChanged(prev: boolean, next: boolean): void {
+        if (super.disabledChanged) {
```

Comment:
This is because this is part of the [`FormAssociated` interface](https://github.com/microsoft/fast/blob/c64110d2f1dfbc24a0978c362bcf72b8958e6d73/packages/web-components/fast-foundation/src/form-associated/form-associated.ts#L121) which types this to a nullable callback.

Reply:
Excellent, thanks for checking 🙏